docs/content/policy-reference.md

```diff
@@ -69,8 +69,14 @@ complex types.
 ### Objects
 | Built-in | Description |
 | -------- | ----------- |
-| <span class="opa-keep-it-together">`filtered := json.filter(object, paths)`</span> | `filtered` is the remaining data from `object` with only keys specified in `paths` which is an array or set of key paths. Each path may be a JSON string path or an array of path segments. For example: `json.filter({"a": {"b": "x", "c": "y"}}, ["a/b"]` will result in `{"a": {"b": "x"}}`). The `json` string `paths` may reference into array values by using index numbers. For example with the object `{"a": ["x", "y", "z"]}` the path `a/1` references `y` |
 | <span class="opa-keep-it-together">`value := object.get(object, key, default)`</span> | `value` is the value stored by the `object` at `key`. If no value is found, `default` is returned. |
+| <span class="opa-keep-it-together">`output := object.remove(object, keys)`</span> | `output` is a new object which is the result of removing the specified `keys` from `object`. `keys` must be either an array or set of keys. |
```

Comment:
I'm tempted to suggest that `keys` should also accept an `object`. The built-in would remove matching keys (i.e., values would be ignored.) Otherwise if you have an object and want to remove matching keys, you'll need to write a comprehension--at that point, you may as well use one to start with.

Reply:
Yea, I was wondering about that... I also though about making it recursive so that for an object it would delete the leaf nodes and leave others behind. Eg:

Instead of:
```
a := {"x": {"y1": "foo", "y2": "bar}, "z": "baz"}
object.remove(a, {"x": {"y1": ""}}) == {"z": "baz"}
```
You could instead get:
```
a := {"x": {"y1": "foo", "y2": "bar"}, "z": "baz"}
object.remove(a, {"x": {"y1": ""}}) == {"x": {"y2": "bar"}"z": "baz"}
```
The reasoning behind this is that right now to support a workflow where you effectively want to re-use some default object in tests. With `object.union` we can merge in new fields, and overwrite keys.. but you can't effectively "delete" a nested key. This remove helps on the sub-object but you can't union it back in to blank out some key in the base version. I was eyeing something like this to avoid having to go full in with `json.patch` which adds in a whole bunch of other stuff when all we needed was to delete some nested field (or maybe we do a limited `json.patch` implementation or just do `json.remove` or something).

Either way I'll start by supporting just the top level keys, I agree its a good change.

Reply:
Latest version adds support the objects, I opted to not do the recursive thing.. staring at it more I decided it was complicating things too much.